.github/workflows/buildAndTest.yml

```diff
@@ -99,10 +79,60 @@ jobs:
     # Publishing is necessary to make the release visible to `pip`
     # on the github releases page.
     - name: Publish Release (if requested)
-      if: github.event.inputs.release_id != ''
       id: publish_release
+      if: github.event.inputs.release_id != ''
       uses: eregon/publish-release@v1
       env:
         GITHUB_TOKEN: ${{ secrets.WORKFLOW_INVOCATION_TOKEN }}
       with:
         release_id: ${{ github.event.inputs.release_id }}
+
+  build-out-of-tree:
```

Comment:
Do we really want to re-build everything twice on every commit and re-run all the tests?

I think that for now it would be good to do this as a scheduled job that runs post-commit or nightly/N-times-per-day.

Reply:
Thanks for your feedback, you're right that this could become a problem. I think I can change that to run on push (ie on merge). Another option would be to disable tests in this second build, since they're probably identical, and we're just testing whether cmake builds everything correctly. IMHO, keeping the build running in PRs provides for better feedback, as otherwise you could find out a change breaks main only after it was merged. Wdyt?

Reply:
I think if we can reduce the time taken to just building Torch-MLIR (with the LLVM build cached almost always) and running the check-torch-mlir-all, then it would be okay to run on every PR.

Reply:
There was a caching problem with the PR at the time you reviewed, which is fixed in [c579ff0](https://github.com/llvm/torch-mlir/pull/738/commits/c579ff0ec42c6df9946aa955a88d1de5bb9fff9d). See eg https://github.com/ljfitz/torch-mlir/actions/runs/2162118862 for a trial run, which shows that the total build time is roughly equivalent to the current build time. For the in-tree job, it should be identical. The out-of-tree job takes a couple of minutes longer but runs in parallel.

Reply:
Note that the first out-of-tree build on these commits will probably take > an hour because it's running without a cache